hooks/pre/12-clear_pulp_content.rb

```diff
@@ -0,0 +1,45 @@
+require 'fileutils'
+
+def clear_pulp_content
+  content_dir = '/var/lib/pulp/content'
+  if File.directory?(content_dir)
+    logger.debug "Removing Pulp content inside '#{content_dir}'."
+    FileUtils.rm_rf(Dir.glob(File.join(content_dir, '*')))
+    logger.debug "Removed Pulp content inside '#{content_dir}', now resetting Pulp Repo importers."
+    mongo_config = load_mongo_config
+    start_services(['rh-mongodb34-mongod']) unless remote_host?(mongo_config[:host])
+    reset_repo_importers(mongo_config)
+    logger.info 'Pulp content successfully removed.'
+  else
+    logger.warn "Pulp content directory not present at '#{content_dir}'."
+  end
+end
+
+def reset_repo_importers(config)
+  if config[:ssl]
+    ssl = "--ssl"
+    if config[:ca_path]
+      ca_cert = "--sslCAFile #{config[:ca_path]}"
+      client_cert = "--sslPEMKeyFile #{config[:ssl_certfile]}" if config[:ssl_certfile]
+    end
+  end
+  username = "-u #{config[:username]}" if config[:username]
+  password = "-p #{config[:password]}" if config[:password]
+  host = "--host #{config[:host]} --port #{config[:port]}"
+  cmd_base = '--eval \'db.repo_importers.update({"scratchpad": {$ne: null}}, {$set: {"scratchpad.repomd_revision": null}}, {"multi":true})\''
+  cmd = "mongo #{config[:database]} #{username} #{password} #{host} #{ssl} #{ca_cert} #{client_cert} #{cmd_base}"
+  execute(cmd)
+end
+
+if app_value(:clear_pulp_content)
+  if !pulp_enabled?
+    logger.warn 'Skipping Pulp content reset because Pulp feature is disabled in scenario answers.'
+  elsif app_value(:reset_data)
+    logger.warn "Skipping Pulp content reset because all Pulp data will be reset with '\--reset-data\'."
```

Comment:
Perhaps just make this empty. The user doesn't need to be warned about this IMHO.
```suggestion
```

Reply:
@ekohl If we are going to prevent `--clear-pulp-content` from running when using `--reset-data` then I think it absolutely makes sense to warn the user that using them together doesn't exactly make sense to do.

Then, I just noticed that `--reset-data` clears out `/var/lib/pulp/{distributions,published,repos}/*` but not `/var/lib/pulp/content`...

So either that directory should be included as well (I believe that is the case), or we shouldn't actually prevent both from running.

The way I'm thinking about it right now, I lean towards leaving the warning and opening another PR to make sure `--reset-data` also clears `/var/lib/pulp/content`

Do you agree @ekohl ?

Reply:
sing them together is redundant, but IMHO not a real issue. Perhaps the way to deal with this is to skip the database clean if `--reset-data` is also passed in?

Another option is to completely ignore it. It's been like this for ages and planned to be removed. Rather than trying to figure out how to deal with it properly, I'd be ok with ignoring it.

Reply:
@ekohl I think the handling already implemented in this PR becomes correct if `--reset-data` also clears `/var/lib/pulp/content`

I also think that `--reset-data` should also clear `/var/lib/pulp/content` given that it resets the pulp database completely and clears other directories in `/var/lib/pulp`... this seems like a bug to me that we should fix.

Do you disagree with either point?

Reply:
I wonder if `--reset-data` should clear the content. I've never used `--reset-data` nor stored any data in Pulp so I don't know what the expectation is. Perhaps Pulp can pick up content that's already on disk to prevent resyncing but doesn't publish it in any way?

Reply:
The expectation and use of this has been to clear the database and all content on disk. I think if Pulp 3 can handle clearing database and leaving content to be smartly picked up we could implement an additional tweak to the hook but shouldn't optimize that at this point in time.

Reply:
I confirmed with the Katello team it is indeed a bug -- there is no reason to preserve `/var/lib/pulp/content/*` when clearing the Pulp database -- so I opened https://projects.theforeman.org/issues/30776 and the fix with #572

Reply:
Ok, so now #572 is merged I think we can drop this warning. It's now purely a redundant parameter when the user passes it in but that's fine. No need for any warning. Note I'm **not** saying you should remove `elsif app_value(:reset_data)`, only the warning. You can also downgrade it to a debug message.

Reply:
OK, I changed the log level of the message to debug.